server/src/main/scala/de/frosner/broccoli/controllers/InstanceController.scala

```diff
@@ -51,54 +52,28 @@ case class InstanceController @Inject()(instanceService: InstanceService, overri
     }
   }
 
-  def create = StackAction { implicit request =>
-    val maybeValidatedInstanceCreation = request.body.asJson.map(_.validate[InstanceCreation])
-    maybeValidatedInstanceCreation
-      .map { validatedInstanceCreation =>
-        validatedInstanceCreation
-          .map { instanceCreation =>
-            InstanceController.create(instanceCreation, loggedIn, instanceService) match {
-              case InstanceCreationSuccess(creation, instanceWithStatus) =>
-                Results
-                  .Status(201)(Json.toJson(instanceWithStatus))
-                  .withHeaders(
-                    HeaderNames.LOCATION -> s"/api/v1/instances/${instanceWithStatus.instance.id}" // TODO String constant
-                  )
-              case InstanceCreationFailure(creation, reason) =>
-                Results.Status(400)(s"Creating $instanceCreation failed: $reason")
-            }
-          }
-          .getOrElse(Results.Status(400)("Expected JSON data"))
-      }
-      .getOrElse(Results.Status(400)("Expected JSON data"))
+  def create = StackAction(parse.json[InstanceCreation]) { implicit request =>
+    InstanceController
+      .create(request.body, loggedIn, instanceService)
+      .fold(
+        _.toHTTPResult,
```

Comment:
The `@typeclass` annotation generates boilerplate for type classes (see [README](https://github.com/mpilquist/simulacrum)), including an `ops` object which provides syntax support (hence the "ops" import above).  Essentially the "ops" object turns every method of the type class into an extension method on values that have an instance of the type class available.

And indeed, `InstanceError` has a `ToHTTPResult` instance in its companion object.